File: Binary Trees/flatten_binary_tree.py
class Binary_Tree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def flatten_binary_tree(tree):
    in_order = in_order_traversal(tree, [])
    for i in range(len(in_order) - 1):
        left = in_order[i]
        right = in_order[i + 1]
        left.right = right
        right.left = left
    return in_order[0]


def in_order_traversal(tree, array):
    if tree != None:
        in_order_traversal(tree.left, array)
        array.push(tree)
        in_order_traversal(tree.right, array)
    return array

File: Binary Trees/flattenBinaryTree.js
class BinaryTree {
    constructor(value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

const flattenBinaryTree = (root) => {
    const inOrder = inOrderTraversal(root, []);
    for (let i = 0; i < inOrder.length - 1; i++) {
        const left = inOrder[i]
        const right = inOrder[i + 1]
        left.right = right
        right.left = left
    }
    return inOrder[0];
}

const inOrderTraversal = (tree, array) => {
    if (tree !== null) {
        inOrderTraversal(tree.left, array)
        array.push(tree)
        inOrderTraversal(tree.right, array)
    }
    return array;
}